pymc3/distributions/continuous.py

```diff
@@ -110,27 +115,41 @@ class BoundedContinuous(Continuous):
     """Base class for bounded continuous distributions"""
 
 
+class CircularContinuous(Continuous):
+    """Base class for circular continuous distributions"""
+
+
 @logp_transform.register(PositiveContinuous)
-def pos_cont_transform(op):
+def pos_cont_transform(op, *args):
     return transforms.log
 
 
 @logp_transform.register(UnitContinuous)
-def unit_cont_transform(op):
+def unit_cont_transform(op, *args):
     return transforms.logodds
 
 
 @logp_transform.register(BoundedContinuous)
-def bounded_cont_transform(op):
+def bounded_cont_transform(op, transform_args):
+    if transform_args is None:
+        raise ValueError(f"Must specify transform args for distribution of {op}")
+
     def transform_params(rv_var):
-        _, _, _, lower, upper = rv_var.owner.inputs
+        _, _, _, *args = rv_var.owner.inputs
+        lower = args[transform_args[0]]
+        upper = args[transform_args[1]]
```

Comment:
You could just override `Distribution.__new__` and create the transform (see [`Dirchlet.__new__`](https://github.com/pymc-devs/pymc3/blob/v4/pymc3/distributions/multivariate.py#L390)).

Reply:
In that case do we want to keep the bounded continuous class for anything? Maybe to raise an error if transform is UNSET?